service/header/interface.go

```diff
@@ -33,6 +33,11 @@ type Broadcaster interface {
 // Exchange encompasses the behavior necessary to request ExtendedHeaders
 // from the network.
 type Exchange interface {
+	// Start starts the Exchange lifecycle.
+	Start(ctx context.Context) error
+	// Stop stops the Exchange lifecycle.
+	Stop(ctx context.Context) error
```

Comment:
another nit: this can be omitted to shorten the interface while each implementation can be implemented as they need to, e.g CoreExchange does not need them and some other implementations may have `Close()` error instead.